yubikey-luks-enroll

```diff
@@ -22,6 +22,10 @@ while getopts ":s:d:hcv" opt; do
 		DISK=$OPTARG
 		echo "setting disk to $OPTARG."
 		;;
+	y)
+		YUBIKEY_LUKS_SLOT=$OPTARG
```

Comment:
What could happen is: the user creates his challenge manually in slot 1 without changing ``ykluks.cfg``. Then he will not be able to boot, since the initramfs still tries to authenticate with slot 2.

I am a fan of not overcompllicating things (like trying to avoid any possible mistake and thus adding more code, that can actually break).

But maybe you can add some kind of warning: "You are enrolling slot 1 of your yubikey. Please note, that during boot time slot 2 is configured to be used." if the slots from the command line and from ykluks.cfg differ.

Reply:
I would also initialize `YUBIKEY_LUKS_SLOT` to default value in each script that uses it. When people do upgrades and they changed `ykluks.cfg` then it's possible the  new config would end up only as `.dpkg-new` and variable would be empty.

Reply:
I agree, a warning is definitively useful here. I added one that should be hard to overlook.

Reply:
@Vincent43 good point, fixed.